source/mainnet/smart-contracts/onboarding-guide-ethereum-developers/faq.rst

```diff
@@ -0,0 +1,760 @@
+
+.. dropdown::  How can I test a smart contract?
+
+    You can follow the chapter :ref:`testing a smart contract<piggy-bank-testing>` in the piggy bank tutorial.
+
+.. dropdown:: Can I build tests that print a table of content structure (a layered output similar to mocha and chai tests)?
+
+    Yes. You can use a similar pattern as shown below:
+
+    .. code-block:: console
+
+        #[cfg(test)]
+        mod tests {
+            mod func_a {
+                #[test]
+                fn correct_params_correct_result(){...}
+                #[test]
+                fn invalid_params_invalid_result(){...}
+            }
+        }
+
+    The test cases are printed as follows:
+
+    .. code-block:: console
+
+        running 2 tests
+        test tests::func_a::correct_params_correct_result ... ok
+        test tests::func_a::invalid_params_invalid_result ... ok
+
+.. dropdown::  How are `smart contract addresses` represented on Concordium?
+
+    In terms of naming, Concordium uses `contract`, and `account` to refer
+    to the Ethereum equivalent of a `smart contract`, and an `externally owned address`, respectively.
+    The word `address` refers to either an `account` address or a `contract` address on Concordium.
+
+    Contract addresses on Concordium are represented by an index and a subindex as seen below.
+    When invoking the `init` function, a new smart contract instance is
+    deployed and assigned the next index number in sequential order.
+    The subindex is currently not in use and is always 0. There are plans to give the subindex meaning in the future.
+
+    In contrast to Ethereum, you cannot send CCD to a contract address (or account address) before they have been deployed/initialized on the Concordium chain.
+
+    .. code-block:: console
+
+        ContractAddress {
+            index:    1234,
+            subindex: 0,
+        };
+
+.. dropdown::  How are `externally owned addresses` represented on Concordium?
+
+    In terms of naming, Concordium uses `contract`, and `account` to refer
+    to the Ethereum equivalent of a `smart contract`, and an `externally owned address`, respectively.
+    The word `address` refers to either an `account` address or a `contract` address on Concordium.
+
+    Accounts on the chain are identified via an account address, which is a 32-byte sequence.
+    The address is usually displayed in Base58Check encoding with version byte 1.
+    An example of such an address is 3ZFGxLtnUUSJGW2WqjMh1DDjxyq5rnytCwkSqxFTpsWSFdQnNn.
+
+    In contrast to Ethereum, `accounts` are also deployed on-chain and their corresponding `account` address only exists from that point on.
+    You cannot send CCD to an account address (or a smart contract address) before they have been deployed/initialized on the Concordium chain.
+    When a smart contract tries to interact with an address that has not been deployed/initialized yet, the smart contract will revert on the Concordium chain.
```

Comment:
I don't think this is accurate. The smart contract will be told that the interaction failed and can choose what to do.

Reply:
I changed it to:
When a smart contract tries to interact with an address that has not been deployed/initialized yet, the interaction fails.